components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -80,7 +85,10 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
                         validationResponseDTO.getAuthorizedUser());
 
         String appResideOrgId = getOrganizationIdFromTenantDomain(authorizedUser.getTenantDomain());
-        checkOrganizationIsAllowedToSwitch(appResideOrgId, accessingOrgId);
+        OAuthAppDO oAuthAppDO = (OAuthAppDO) tokReqMsgCtx.getProperty(OAUTH_APP_PROPERTY);
```

Comment:
We can load the OAuth App by this Utill class https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java#L2286

Using dao layer classes of a different module is not a good practice in the context of OSGI.

Reply:
Future improvements.